torch/onnx/verify.py

```diff
@@ -0,0 +1,262 @@
+import copy
+import io
+import os
+import tempfile
+from contextlib import ExitStack
+from typing import Optional, Union
+
+import numpy as np
+
+import torch
+from torch import Tensor
+from torch.onnx.utils import unpack_quantized_tensor
+
+_ORT_PROVIDERS = ["CPUExecutionProvider"]
+
+
+def _flatten_tuples(elem):
+    flattened = []
+    for t in elem:
+        if isinstance(t, tuple):
+            flattened.extend(_flatten_tuples(t))
+        else:
+            flattened.append(t)
+    return flattened
+
+
+def _to_numpy(elem):
+    if isinstance(elem, Tensor):
+        if elem.requires_grad:
+            return elem.detach().cpu().numpy()
+        else:
+            return elem.cpu().numpy()
+    elif isinstance(elem, (list, tuple)):
+        return [_to_numpy(inp) for inp in elem]
+    elif isinstance(elem, bool):
+        return np.array(elem, dtype=bool)
+    elif isinstance(elem, int):
+        return np.array(elem, dtype=int)
+    elif isinstance(elem, float):
+        return np.array(elem, dtype=float)
+    elif isinstance(elem, dict):
+        flattened = []
+        for k in elem:
+            flattened += [_to_numpy(k)] + [_to_numpy(elem[k])]
+        return flattened
+    return elem
+
+
+def _convert_to_onnx(
+    model,
+    model_f: Union[str, io.BytesIO],
+    input=None,
+    opset_version=None,
+    do_constant_folding=True,
+    keep_initializers_as_inputs=True,
+    dynamic_axes=None,
+    input_names=None,
+    output_names=None,
+    fixed_batch_size=False,
+    training=None,
+    verbose=False,
+    ort_providers=_ORT_PROVIDERS,
+    ort_optim_on=True,
```

Comment:
naming: abbr

Reply:
optim seems a reasonable abbreviation for optimization.

Reply:
well, optim is not something new, already used by other packages as well. I don't have any strong preference though, just to bring up.

https://pytorch.org/docs/stable/optim.html
https://www.rdocumentation.org/packages/stats/versions/3.6.2/topics/optim
> As a rule of thumb, an abbreviation is probably OK if it's listed in Wikipedia.